app/models/widget.rb

```diff
@@ -111,8 +111,11 @@ def get_row_count
 
     if (result['data'].kind_of?(Array))
       count = result['data'].first.values.first
-    else
+    elsif (result['data'].is_a?(Hash))
       count = result['data'].values.first
+    else
+      Rails.logger.debug 'Widget.get_row_count - Expected hash or array, got ' + result.inspect
+      count = 0
```

Comment:
looks ok, personally would prefer the assignment form
````
    count = if cond1
        value1
    elsif cond2
        value2
    else
        value3
    end
````
and no parens on conditions :)